### PR TITLE
Add post stage for creating test-report manifest after integ test workflow

### PR DESCRIPTION
**Pull request by zelinh**
### Description
Add the post stage activity to call the groovy library `createUploadTestReportManifest` to create the `test-report` manifest for each time our integ test workflow finishes. 

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/1274

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: jenkins/opensearch-dashboards/integ-test.jenkinsfile [hunk 1]
```diff
@@ -235,6 +235,27 @@ pipeline {
...
+                    script {
+                        node(AGENT_LABEL) {
+                            docker.withRegistry('https://public.ecr.aws/') {
+                                docker.image(docker_images["$distribution"]).inside(docker_args["$distribution"]) {
+                                    unstash "integtest-opensearch-dashboards-$BUILD_NUMBER"
+                                    downloadBuildManifest(
+                                        url: BUILD_MANIFEST_URL_OPENSEARCH,
+                                        path: BUILD_MANIFEST_OPENSEARCH
+                                    )
+                                    createUploadTestReportManifest(
```
Comment (peternied) [inline jenkins/opensearch-dashboards/integ-test.jenkinsfile:247]: I think there is a file on your hard disk that didn't make it into git, can't find a reference to createUploadTestReportManifest

**Review by gaiksaya — CHANGES_REQUESTED** [review]
Can you add assertion test to OpenSearch integtest as well?

File: jenkins/opensearch-dashboards/integ-test.jenkinsfile (hunk 1 above)
Comment (gaiksaya) [reply jenkins/opensearch-dashboards/integ-test.jenkinsfile:247]: You can see the library here: https://github.com/opensearch-project/opensearch-build-libraries/blob/main/vars/createUploadTestReportManifest.groovy

**Comment by zelinh** [comment]
> Can you add assertion test to OpenSearch integtest as well?

I think i have assertion here in this `tests/jenkins/TestOpenSearchIntegTest.groovy`. Looking into the groovy tests fail. It was good locally..

File: jenkins/opensearch-dashboards/integ-test.jenkinsfile (hunk 1 above)
Comment (zelinh) [reply jenkins/opensearch-dashboards/integ-test.jenkinsfile:247]: Yes. We have our library in the above link.

**Review by gaiksaya — APPROVED** [review]

**Pull request merged**